modules/custom/az_core/src/Plugin/ConfigProvider/QuickstartConfigProvider.php

```diff
@@ -27,7 +27,111 @@ class QuickstartConfigProvider extends ConfigProviderBase {
    * {@inheritdoc}
    */
   public function addConfigToCreate(array &$config_to_create, StorageInterface $storage, $collection, $prefix = '', array $profile_storages = []) {
-    // The caller will aready have loaded config for install.
+    // Remove permissions that don't exist.
+    $config_to_create = $this->trimPermissions($config_to_create);
+  }
+
+  /**
+   * Find newly available permissions from profile.
+   *
+   * @param \Drupal\Core\Extension\Extension[] $extensions
+   *   An associative array of Extension objects, keyed by extension name.
+   *
+   * @return array
+   *   A list of the configuration data keyed by configuration object name.
+   */
+  public function findProfilePermissions(array $extensions = []) {
+
+    // Get active configuration to look for users.
+    $existing_config = $this->getActiveStorages()->listAll();
+    // phpcs:ignore
+    $existing_users = array_filter($existing_config, function ($name) {
+      return (strpos($name, 'user.role.') === 0);
+    });
+    $user_config = $this->getActiveStorages()->readMultiple($existing_users);
+
+    // Build list of permissions by providers (eg. module)
+    // @todo Use injection on user.permissions.
+    // @phpstan-ignore-next-line
+    $permissions_definitions = \Drupal::service('user.permissions')->getPermissions();
+    $permissions_by_provider = [];
+    foreach ($permissions_definitions as $key => $permission) {
+      $permissions_by_provider[$permission['provider']][] = $key;
+    }
+    // Filter list only to newly active permissions.
+    $permissions_by_provider = array_intersect_key($permissions_by_provider, $extensions);
+    $new_perms = [];
+    foreach ($permissions_by_provider as $perms) {
+      $new_perms = $new_perms + $perms;
+    }
+    sort($new_perms);
```

Comment:
Potential performance improvement (might not be very significant):

return empty arraay here if no new permissions exist (to prevent the rest of this method from running).
```suggestion
    return [];
    sort($new_perms);
```